fhir-persistence-params/src/main/java/com/ibm/fhir/persistence/params/database/PlainParamValueProcessor.java

```diff
@@ -432,6 +431,12 @@ protected void addMissingCommonTokenValues(List<CommonTokenValue> missing) throw
     public void resolveCanonicalValues(List<CommonCanonicalValue> unresolvedCanonicalValues,
         Map<CommonCanonicalValueKey, CommonCanonicalValue> commonCanonicalValueMap) throws FHIRPersistenceException {
         // identify which values aren't yet in the database
+
+        // Sort the unresolved list so we always process stuff in the same order
+        Collections.sort(unresolvedCanonicalValues, (a,b) -> {
+            return a.getUrl().compareTo(b.getUrl());
```

Comment:
does CommonCanonicalValue (or our logic around it) ensure that the URL is never null?
if not, maybe move this implementation right into the CommonCanonicalValue class?

Reply:
yeah, changed this